### PR TITLE
fix: add social auth tests

### PR DESCRIPTION
**Pull request by markusahlstrand**
Did a refactor to call the userinfo endpoint for the Vipps connection. Added some tests for the social auth flows, but should probably add tests for the actual oauth2 class as it's being mocked now.

### CODE REVIEW

File: src/authentication-flows/social.ts [hunk 1]
```diff
@@ -69,6 +70,54 @@ export interface socialAuthCallbackParams {
   code: string;
 }
 
+async function getProfile(token: TokenResponse) {
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:73]: Refactored out the getProfile function

File: src/authentication-flows/social.ts [hunk 2]
```diff
@@ -69,6 +70,54 @@ export interface socialAuthCallbackParams {
...
+    } = parseJwt(token.id_token);
+
+    if (emailRaw) {
+      const email = emailRaw.toLocaleLowerCase();
+      return { sub, email, email_verified, profileData };
+    }
+  }
+
+  const { iss, sub } = parseJwt(token.access_token);
+  const userinfoResponse = await fetch(`${iss}/userinfo`, {
```
Comment (markusahlstrand) [inline src/authentication-flows/social.ts:100]: Make a call to the userInfo endpoint if there's no email in the id-token. I think this should be a setting on the connection..

File: test/fixtures/oauth2Client.ts [hunk 3]
```diff
@@ -25,9 +27,42 @@ export class OAuth2ClientFixture implements IOAuth2Client {
   }
 
   async exchangeCodeForTokenResponse(code: string): Promise<TokenResponse> {
+    const access_token = await createToken({
```
Comment (markusahlstrand) [inline test/fixtures/oauth2Client.ts:30]: Add propert tokens to the fixtures

File: src/authentication-flows/social.ts (hunk 1 above)
Comment (danielgent) [reply src/authentication-flows/social.ts:73]: How is it refactoring though if the tests are failing :smile:

**Comment by danielgent** [comment]
What does this add over the social hono tests that I have?

I would imagine we want to be using hono/testing instead of the isolated unit tests, *and* this test is another dependency on the in-memory data adapters

I haven't looked at the :x: on the CI/CD because I don't think it's helping you as a developer :joy: 
* do one thing
* push this up to a  PR
* if the tests pass, do another thing, if not, see why
:wink:

**Comment by markusahlstrand** [comment]
This is fixed in another branch

**Pull request closed without merging**